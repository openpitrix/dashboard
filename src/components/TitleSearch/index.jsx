import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { noop } from 'lodash';

import { Input } from 'components/Base';

import styles from './index.scss';

@translate()
export default class AppStatistics extends Component {
  static propTypes = {
    onClear: PropTypes.func,
    onSearch: PropTypes.func,
    placeholder: PropTypes.string,
    searchWord: PropTypes.string,
    title: PropTypes.string
  };

  static defaultProps = {
    title: '',
    searchWord: '',
    onSearch: noop,
    onClear: noop,
    placeholder: ''
  };

  render() {
    const {
      onSearch, onClear, searchWord, placeholder, title
    } = this.props;

    return (
      <div className={styles.titleSearch}>
        {title}
        <Input.Search
          className={styles.search}
          placeholder={placeholder}
          value={searchWord}
          onSearch={onSearch}
          onClear={onClear}
        />
      </div>
    );
  }
}
