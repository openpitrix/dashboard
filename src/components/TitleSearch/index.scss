@import '~scss/vars';

.titleSearch {
  @include title-font($font-size: 16px);
  margin: 18px 0;

  .search {
    float: right;
    line-height: 32px;
    font-size: 12px;

    input {
      background-color: $background-color;
      border-color: $background-color;

      &:hover {
        border-color: $N10;
      }

      &:focus {
        border-color: $P75;
      }
    }

    svg {
      float: right;
    }
  }
}
