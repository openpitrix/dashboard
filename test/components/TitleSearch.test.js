import TitleSearch from 'components/TitleSearch';

describe('Component TitleSearch', () => {
  it('basic render', () => {
    const wrapper = render(<TitleSearch />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
