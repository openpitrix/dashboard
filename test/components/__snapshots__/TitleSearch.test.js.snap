// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TitleSearch basic render 1`] = `
<div
  class="titleSearch"
>
  <div
    class="inputGroup search"
  >
    <span
      class="icon icon-search"
      style="width:16px;height:16px"
    >
      <svg
        class="qicon qicon-magnifier qicon-dark"
      >
        <use
          href="#qui-magnifier"
        />
      </svg>
    </span>
    <input
      class="input"
      placeholder=""
      type="text"
      value=""
    />
  </div>
</div>
`;
